### PR TITLE
School Library: Preserve data

### PR DESCRIPTION
**Pull request by Emmy-github-webdev**
# School Library: Preserve data

In this project @Ntare22  and @Emmy-github-webdev 
- Create rentals.json, books.json, and rentals.json file
- Save files to the respective JSON files
- Fetch data from the file

### CODE REVIEW

File: logic/logic_io.rb [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  end
+
+  def create_people(person)
+    if person['specialization']
+      Teacher.new(person['specialization'], person['name'], person['age'], person['id'])
+    else
+      Student.new(person['age'], person['name'], person['classroom'],
+                  person['id'], parent_permission: person['parent_permission'])
+    end
+  end
```
Comment (omar-labana) [inline logic/logic_io.rb]: - [ ] Creating a student or a teacher makes this error arise on saving, kindly solve it.
image

**Review by omar-labana — CHANGES_REQUESTED** [review]
Hi @Emmy-github-webdev and @Ntare22, 

Excellent work so far! 👌
You have implemented almost all the required features correctly. ✔️ 
There are some cases that you need to adjust but you are almost done! 😀

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by hiromataba — APPROVED** [review]
# Status: Approved :tada:  :champagne:  :confetti_ball: 
Hi  :wave:   :wave: @Emmy-github-webdev  

200w 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### To Highlight :1st_place_medal: 
- Your Pull Request has a proper title and description. ✔️  
- No Linter errors :heavy_check_mark: 
- Your app is stroring  data :+1: 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: logic/logic_io.rb (hunk 1 above)
Comment (Emmy-github-webdev) [reply logic/logic_io.rb]: Thanks @omar-labana

**Comment by Emmy-github-webdev** [comment]
Thanks @hiromataba

**Pull request merged**